pkg/galley/galley.go

```diff
@@ -16,48 +16,118 @@ package galley
 
 import (
 	"context"
+	"fmt"
+	"net/url"
+	"os"
+	"time"
 
 	"google.golang.org/grpc"
 	mcpapi "istio.io/api/mcp/v1alpha1"
 
-	"istio.io/istio/pkg/log"
+	"istio.io/pkg/log"
 
+	"github.com/maistra/ior/pkg/bootstrap"
 	"github.com/maistra/ior/pkg/route"
 	networking "istio.io/api/networking/v1alpha3"
 	_ "istio.io/istio/galley/pkg/metadata" // Import the resource package to pull in all proto types.
-	mcpclient "istio.io/istio/pkg/mcp/client"
+	"istio.io/istio/pkg/mcp/creds"
 	"istio.io/istio/pkg/mcp/sink"
 	"istio.io/istio/pkg/mcp/testing/monitoring"
 )
 
-// ConnectToGalley ...
-func ConnectToGalley(galleyAddr string) {
-	ctx := context.Background()
-	conn, err := grpc.DialContext(ctx, galleyAddr, grpc.WithInsecure())
+const (
+	requiredCertCheckFreq = 1 * time.Second
+)
+
+// Galley is responsible to interact with Galley server
+type Galley struct {
+	args *bootstrap.Args
+}
+
+// New returns a Galley instance or an error
+func New(args *bootstrap.Args) (*Galley, error) {
+	if err := args.Validate(); err != nil {
+		return nil, fmt.Errorf("error validating arguments: %v", err)
+	}
+
+	log.Infof("Started IOR with\n%v", args)
```

Comment:
Should the new line be after the args?

Reply:
No. That's the pattern of Istio initialization logs, all startup arguments go in a new line, after this `Started <component> with`